	Installation paths for TL and standalone builds.

	For TL we do not want to substitute the value of $(libdir).

	ToDo:
	Determination of lib_dir and modules_dir in the xindy perl script.

diff -ur -x Makefile.in -x autom4te.cache xindy.orig/configure.ac xindy/configure.ac
--- xindy.orig/configure.ac	2009-12-03 11:52:02.000000000 +0100
+++ xindy/configure.ac	2009-12-03 11:59:47.000000000 +0100
@@ -35,6 +35,11 @@
 # programs, automake's defaults are not usable for us, too.
 AM_INIT_AUTOMAKE([foreign no-define])
 
+# Installation directories for standalone xindy.
+memdir='$(pkglibdir)'
+AC_SUBST([memdir])
+AM_CONDITIONAL([TEXLIVE_BUILD], [false])
+
 # Test for building make-rules
 AC_ARG_ENABLE(make-rules, AS_HELP_STRING([--enable-make-rules ],[build and install make-rules package (default is YES)]),[BUILDRULES=$enableval],[BUILDRULES=yes])
 test "$BUILDRULES" = "yes"  &&  AC_PROG_LATEX
@@ -56,6 +61,7 @@
 # FIXME: lex actually must really be flex. But there is no AC_PROG_FLEX!?
 AC_PROG_LEX
 AC_PROG_CC
+AC_PROG_LN_S
 
 # Check system type
 AC_CANONICAL_HOST
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/src/Makefile.am xindy/src/Makefile.am
--- xindy.orig/src/Makefile.am	2009-12-03 11:53:59.000000000 +0100
+++ xindy/src/Makefile.am	2009-12-03 12:00:48.000000000 +0100
@@ -26,7 +26,7 @@
 
 # Automake has no CLISP support; moreover xindy.mem is not a binary
 # to be stripped upon 'make install-strip'.  We define it as DATA.
-pkglib_DATA = xindy.mem
+mem_DATA = xindy.mem
 
 # Tell the dist target about our sources, define the object files for
 # our own rule and the .lib files for the clean definition. Don't use
diff -ur -x Makefile.in -x autom4te.cache xindy.orig/user-commands/Makefile.am xindy/user-commands/Makefile.am
--- xindy.orig/user-commands/Makefile.am	2009-12-03 01:31:28.000000000 +0100
+++ xindy/user-commands/Makefile.am	2009-12-08 22:55:09.000000000 +0100
@@ -20,10 +20,47 @@
 ## Free Software Foundation, Inc., 51 Franklin Street, Fifth Floor,
 ## Boston, MA 02110-1301 USA.
 
-bin_SCRIPTS = texindy xindy xindy.v2
+scripts = texindy xindy
+
+bin_SCRIPTS =
+
+if TEXLIVE_BUILD
+noinst_SCRIPTS = $(scripts)
+scriptspath = texmf/scripts/xindy
+scriptsdir = ${prefix}/$(scriptspath)
+install-exec-hook: $(scripts)
+	$(MKDIR_P) "$(DESTDIR)$(scriptsdir)"
+	@for f in $(scripts); do \
+	  $(INSTALL_SCRIPT) `test -f "$$f" || echo "$(srcdir)/"`$$f $(DESTDIR)$(scriptsdir)/$$f.pl; \
+	done
+	case "$(bindir)" in \
+	  */bin) $(MAKE) $(AM_MAKEFLAGS) REL=.. install-links;; \
+	  */bin/*) $(MAKE) $(AM_MAKEFLAGS) REL=../.. install-links;; \
+	  *) echo "strange directory '$(bindir)' for linked scripts" >&2; \
+	     exit 1;; \
+	esac
+.PHONY: install-links
+install-links:
+	@cd $(DESTDIR)$(bindir) && \
+	for f in $(scripts); do \
+	  rm -f $$f; \
+	  echo "creating link '$$f -> '$(REL)/$(scriptspath)/$$f.pl"; \
+	  $(LN_S) $(REL)/$(scriptspath)/$$f.pl $$f; \
+	done
+uninstall-hook:
+	@for f in $(scripts); do \
+	  rm -f $(DESTDIR)$(bindir)/$$f; \
+	  rm -f $(DESTDIR)$(scriptsdir)/$$f.pl; \
+	done
+
+xindy: xindy.in
+	cat $(srcdir)/xindy.in >xindy
+else !TEXLIVE_BUILD
+bin_SCRIPTS += $(scripts) xindy.v2
 
 xindy: xindy.in Makefile
 	sed -e 's:@libdir\@:$(libdir):' $(srcdir)/xindy.in >xindy
+endif !TEXLIVE_BUILD
 
 xindy.v2: xindy.v2.in
 	cat $(srcdir)/xindy.v2.in >xindy.v2
